docs/cheat-sheets/javascript-code-injection.mdx

```diff
@@ -0,0 +1,95 @@
+---
+slug: javascript-code-injection
+---
+
+import LinkToRegistryRule from "/src/components/LinkToRegistryRule"
+import CodeInjectionIntro from "/src/components/concept/_code-injection-intro.mdx"
+
+# Code injection prevention for JavaScript
+
+<CodeInjectionIntro />
+
+#### Check your project for these conditions:
+
+```bash
+semgrep scan --config="r/javascript.lang.security.audit.vm-injection.vm-runincontext-context-injection" --config="r/javascript.lang.security.detect-eval-with-expression.detect-eval-with-expression"
+```
+
+---
+
+## 1. Executing or evaluating code
+
+### 1.A. running code with VM module
+
+The `vm` module enables compiling and running code within V8 Virtual Machine contexts. The `vm` module is not secure. Do not use it to run untrusted code. If code passed to `vm` functions is controlled by user input it could result in code injection. See [VM (executing JavaScript)](https://nodejs.org/dist/latest-v18.x/docs/api/vm.html#vm-executing-javascript) documentation.
```

Comment:
```suggestion
The `vm` module enables compiling and running code within V8 Virtual Machine contexts. The `vm` module is not secure. Do not use it to run untrusted code. If user input is used as part of the code passed to `vm` functions, it could result in code injection. See [VM (executing JavaScript)](https://nodejs.org/dist/latest-v18.x/docs/api/vm.html#vm-executing-javascript) documentation.
```

Reply:
```suggestion
The `vm` module enables compiling and running code within V8 Virtual Machine contexts. The `vm` module is not secure. Do not use it to run untrusted code. If user input is used as a part of the code passed to `vm` functions, it can result in code injection. See [VM (executing JavaScript)](https://nodejs.org/dist/latest-v18.x/docs/api/vm.html#vm-executing-javascript) documentation.
```